manifests/agent.pp

```diff
@@ -0,0 +1,51 @@
+class ss_nessus_agent::agent inherits ::ss_nessus_agent {
+
+  if $ss_nessus_agent::http_proxy {
+      $proxy_environment = ["http_proxy=${ss_nessus_agent::http_proxy}", "https_proxy=${ss_nessus_agent::http_proxy}"]
+  } else {
+      $proxy_environment = []
+  }
+
+  exec { 'download':
+    command     => "curl -s -f ${ss_nessus_agent::download_url} -o /usr/src/nessusagent.deb",
+    path        => '/usr/bin:/usr/sbin:/bin',
+    onlyif      => "test ! -f /usr/src/nessusagent.deb",
+    environment => $proxy_environment,
+  }
+  -> package { 'nessusagent':
+    ensure   => installed,
+    provider => dpkg,
+    source   => '/usr/src/nessusagent.deb',
+  }
+
+  # The setup and configuration utilises the custom facts
+  if $params::nessus_agent_enabled == true {
```

Comment:
There are no `$params` in this repostory. Would recommend using a `$enabled` flag on the `init` instead as this currently only works for CCL.